### PR TITLE
REF/TST: use pytest builtin monkeypatch fixture and remove mock fixture

### PR DESCRIPTION
**Pull request by simonjayhawkins**
- [n/a ] xref https://github.com/pandas-dev/pandas/pull/24557#issuecomment-451174496
- [x] tests added / passed
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [n/a ] whatsnew entry

### CODE REVIEW

File: pandas/tests/dtypes/test_inference.py [hunk 1]
```diff
@@ -235,7 +235,6 @@ class MockFile(object):
     # Iterator but no read / write attributes
     data = [1, 2, 3]
     assert not is_file(data)
-    assert not is_file(mock.Mock())
```
Comment (WillAyd) [inline pandas/tests/dtypes/test_inference.py]: What was this changed for?

File: pandas/tests/io/formats/test_format.py [hunk 2]
```diff
@@ -305,16 +305,14 @@ def test_repr_non_interactive(self):
...
-        # This is being backported, so doing it directly here.
+    def test_repr_truncates_terminal_size(self, monkeypatch):
+        # see gh-21180
 
         terminal_size = (118, 96)
-        p1 = mock.patch('pandas.io.formats.console.get_terminal_size',
-                        return_value=terminal_size)
-        p2 = mock.patch('pandas.io.formats.format.get_terminal_size',
-                        return_value=terminal_size)
+        monkeypatch.setattr('pandas.io.formats.console.get_terminal_size',
```
Comment (WillAyd) [inline pandas/tests/io/formats/test_format.py]: Does this reset the attribute after the test or do we need to use the context manager?

File: pandas/tests/io/parser/test_common.py [hunk 3]
```diff
@@ -1814,13 +1814,16 @@ class InvalidBuffer(object):
         parser.read_csv(InvalidBuffer())
 
 
-def test_invalid_file_buffer_mock(all_parsers, mock):
+def test_invalid_file_buffer_mock(all_parsers):
     # see gh-15337
     parser = all_parsers
     msg = "Invalid file path or buffer object type"
 
+    class Mock():
```
Comment (WillAyd) [inline pandas/tests/io/parser/test_common.py]: Personally would rather just name this `Foo` so as not to cause confusion with the real `Mock` class

**Review by WillAyd — CHANGES_REQUESTED** [review]

File: pandas/tests/io/formats/test_format.py (hunk 2 above)
Comment (simonjayhawkins) [reply pandas/tests/io/formats/test_format.py]: @WillAyd the context manager is only required when patching might break `pytest` or where it is required to undo a patch before the test ends. the `monkeypatch` fixture will undo all patches on test teardown.

File: pandas/tests/dtypes/test_inference.py (hunk 1 above)
Comment (simonjayhawkins) [reply pandas/tests/dtypes/test_inference.py]: this is the only use of the `mock` fixture within this test so probably unnecessary now.

**Comment by jreback** [comment]
@simonjayhawkins in followup could add a check for ``mock`` in the ``ci/code_checks`` (meaning we should *not* find it)

**Comment by jreback** [comment]
thanks!

**Pull request merged**